### PR TITLE
ENG-306 - move new dashboard value from localStorage to user/features

### PR DESCRIPTION
**Pull request by adamkecskes**

### CODE REVIEW

**Review by nwinter — CHANGES_REQUESTED** [review]
This should go into some other config than `aceConfig`; `aceConfig` is for code-editor-related settings (Ace being the name of the code editor).

As to where to put it, I think a top-level user property is fine, maybe [here](https://github.com/codecombat/codecombat/blob/cf136d43a175a39511c50de5475a40b1ec521d2a/app/schemas/models/user.js#L404-L404) next to some other Boolean teacher dashboard properties.

As far as values go: `undefined` will currently mean "show old dashboard" and later will mean "show new dashboard", `false` will mean they've explicitly chosen to use the old dashboard, and `true` will mean they've explicitly chosen to use the new dashboard.

We _could_ also have it be, not boolean, but an enum string property, so that we could distinguish whether this is the first time we've shown them the new dashboard by default, so that we can show a first-time message highlighting what has changed and how to go back. Or we could store that extra temporary flag in local storage.

**Review by mrfinch — COMMENTED** [review]
could use `features` object in user for this too.

**Comment by adamkecskes** [comment]
I've moved the flag into a separate property. (`features` didn't really work for me)
It is an enum now.

File: app/schemas/models/user.js [hunk 1]
```diff
@@ -202,7 +202,7 @@ _.extend(UserSchema.properties, {
       })
     }),
 
-  aceConfig: c.object({ default: { language: 'python', keyBindings: 'default', invisibles: false, indentGuides: false, behaviors: false, liveCompletion: true } }, {
+  aceConfig: c.object({ default: { language: 'python', keyBindings: 'default', invisibles: false, indentGuides: false, behaviors: false, liveCompletion: true, newDashboard: false } }, {
```
Comment (mrfinch) [inline app/schemas/models/user.js:205]: remove

File: app/schemas/models/user.js [hunk 2]
```diff
@@ -402,6 +402,11 @@ _.extend(UserSchema.properties, {
   seenNewDashboardModal: { type: 'boolean', description: 'Whether the user has seen the new dashboard onboarding modal? Set to true after the modal is seen and closed by the user' }, // Ozaria
   closedNewTDGetStartedTooltip: { type: 'boolean', description: 'Whether the user has closed the get started tooltip in the new dashboard? Set to true once the user has dismissed the tooltip' }, // Ozaria
 
+  dashboardVersion: {
```
Comment (mrfinch) [inline app/schemas/models/user.js:405]: I dont think keeping it at top-level is a good idea. What issue did you face with features? Could create a new property `flags` if you need. We have properties like this: `seenNewDashboardModal`, `closedNewTDGetStartedTooltip` already which we should club into a single object(later work)

File: app/schemas/models/user.js [hunk 3]
```diff
@@ -402,6 +402,11 @@ _.extend(UserSchema.properties, {
...
+    type: 'string',
+    enum: ['default', 'old-coco', 'merged'],
```
Comment (mrfinch) [inline app/schemas/models/user.js:407]: why do we need default? Won't null be treated as default?

**Comment by adamkecskes** [comment]
@mrfinch I'm not really sure what was my issue with features last time, but now it works...
Backend PR: https://github.com/codecombat/codecombat-server/pull/834

File: app/schemas/models/user.js (hunk 1 above)
Comment (mrfinch) [reply app/schemas/models/user.js:205]: you missed this?

File: app/models/User.js [hunk 4]
```diff
@@ -160,6 +160,10 @@ module.exports = (User = (function () {
     isSmokeTestUser () { return User.isSmokeTestUser(this.attributes) }
     isIndividualUser () { return !this.isStudent() && !User.isTeacher(this.attributes) }
 
+    isNewDashboardActive () {
+      return me.get('features')?.isNewDashboardActive
```
Comment (mrfinch) [inline app/models/User.js:164]: this.get('features')?.isNewDashboardActive

**Review by mrfinch — APPROVED** [review]

**Pull request merged**